libraries/AP_Param/AP_Param.cpp

```diff
@@ -242,12 +239,16 @@ bool AP_Param::check_group_info(const struct AP_Param::GroupInfo *  group_info,
 // check for duplicate key values
 bool AP_Param::duplicate_key(uint16_t vindex, uint16_t key)
 {
-    for (uint16_t i=vindex+1; i<_num_vars; i++) {
-        uint16_t key2 = _var_info[i].key;
-        if (key2 == key) {
-            // no duplicate keys allowed
-            return true;
+    var_table *list = &_list;
+    while (list != nullptr) {
```

Comment:
please use for() structure for list walk

Reply:
done